### ISSUE TITLE
rpc annotation channel only supported with 4 arguments

### ISSUE THREAD

**Issue by KennethWilke** [godotengine/godot#63208]
### Godot version

v4.0.alpha12.official.2c11e6d9e

### System information

Ubuntu 22.04 LTS

### Issue description

I've been tinkering with the RPC changes in 4, and ran into some issues when trying to select the channel in the `@rpc` annotation. When I attempt to set the channel as [described in the blog post](https://godotengine.org/article/multiplayer-changes-godot-4-0-report-2), I get parsing errors stating that string values were expected unless I have exactly 4 arguments and my channel is the last.

### Expected behavior

I was expecting to be able to provide an annotation like `@rpc(unreliable_ordered, 1)` and have the last value taken as the channel ID int. 

### Observed behavior

GDScript fails to parse:

`SCRIPT ERROR: Parse Error: Expected String as argument 2 of annotation "@rpc".`

### Other notes

I looked into the implementation to try to find out more and found [this error handler](https://github.com/godotengine/godot/blob/eea14a0edc3f2a8ab4a91d1e142d48a5348584bb/modules/gdscript/gdscript_parser.cpp#L3880) that states:
`At most 4 arguments are allowed, where only the last argument can be an integer to specify the channel.` From what I'm seeing, it appears like to provide the channel it **must** be `the 4th argument` rather than `the last argument`.

### Workaround

With that in mind, I found I could work around this by providing additional RPC arguments, either by explicitly choosing the default options I wanted or by repetition:

Sane, more verbose/explicit workaround:
```gdscript
@rpc(any_peer, call_remote, reliable, 1)
func my_chat_func():
	print("RPC received on channel 1.")
```

Crazy looking, but allows me to leave other options at their default:
```gdscript
@rpc(any_peer, any_peer, any_peer, 1)
func my_chat_func():
	print("RPC received on channel 1.")
```

### Issue reproduction

`rpc_test.gd`:
```gdscript
extends MainLoop

func _init():
	print("script ran!")
	OS.kill(0)

@rpc(any_peer, 1)
func my_chat_func():
	print("RPC received on channel 1.")
```

Trying to run this via `godot -s`, replicates the error I wasn't expecting:

```
$ godot -s rpc_test.gd 
Godot Engine v4.0.alpha12.official.2c11e6d9e - https://godotengine.org
MESA-INTEL: warning: Performance support disabled, consider sysctl dev.i915.perf_stream_paranoid=0

Vulkan API 1.2.0 - Using Vulkan Device #1: NVIDIA - Quadro M2200
 
SCRIPT ERROR: Parse Error: Expected String as argument 2 of annotation "@rpc".
          at: GDScript::reload (res://rpc_test.gd:8)
ERROR: Method/function failed. Returning: ERR_PARSE_ERROR
   at: reload (modules/gdscript/gdscript.cpp:870)
```

### Workaround reproduction

`rpc_channel_workaround.gd`:
```gdscript
extends MainLoop

func _init():
	print("script ran!")
	OS.kill(0)

@rpc(any_peer, call_remote, reliable, 1)
func my_chat_func():
	print("RPC received on channel 1.")
```
```
$ godot -s rpc_workaround.gd 
Godot Engine v4.0.alpha12.official.2c11e6d9e - https://godotengine.org
MESA-INTEL: warning: Performance support disabled, consider sysctl dev.i915.perf_stream_paranoid=0

Vulkan API 1.2.0 - Using Vulkan Device #1: NVIDIA - Quadro M2200
 
script ran!
Killed
```

**Comment by KennethWilke** [comment]
I looked into this a bit more, and tried experimenting with a few fixes but nothing panned out. I believe this is due to the way the annotation structure is defined in [/modules/gdscript/gdscript_parser.cpp](https://github.com/godotengine/godot/blob/ae5668f81e9ed7ae9e9a13b120d89de884b718be/modules/gdscript/gdscript_parser.cpp#L145).

I tried to see if there were some ways to make the [`PropertyInfo` struct](https://github.com/godotengine/godot/blob/master/core/object/object.h#L149) more flexible with the `Variant::Type type = Variant::NIL;` member, but no obviously good option came to mind.

The first options that came to mind were to use binary flag enum similar to how [`AnnoationInfo::TargetKind`](https://github.com/godotengine/godot/blob/master/modules/gdscript/gdscript_parser.h#L1250) does it. Changing [`Variant::Type`](https://github.com/godotengine/godot/blob/master/core/variant/variant.h#L77) to support this seems like too impactful of a change. Changing `PropertyInfo` to use a different `type` type also feels like possibility but also feels excessive for fixing a single annotation.

This annotation looks like the one oddball that has an API goal of supporting different types in different places. I don't see a current way for the annotations to accept mixed signatures likes `[String, String, String, Int]` and `[String, Int]` but I like the reduced syntax of being able to leave the `mode`, `sync` and `transfer_mode` as default and specifying only the non-defaults I'm wanting.

The most reasonable / least impactful option that comes to mind is to change the `@rpc` annotation `PropertyInfo` to be a `Variant::Dictionary` type. This would only change this one annotation in an API-breaking way, I'm thinking it might allow something like:

```gdscript
@rpc({"mode": "any_peer", "transfer_channel": 1})
func my_chat_func():
	print("RPC received on channel 1.")
```

I'm not sure how feasible/desirable that direction would be, but trying to contribute because I love the hell out of Godot :smile:
